### PR TITLE
feat(console): sign up and sign in change alert

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): sign up and sign in change alert

## Testing

image

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/utilities.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    }
+
+    return { mutation: 'none', value: target };
+  });
+
+  return socialTargetsDiff
+    .slice()
+    .sort(({ value: leftValue }, { value: rightValue }) =>
+      stringCompareFunction(leftValue, rightValue)
+    );
```
Comment (charIeszhao) [inline packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/utilities.ts:157]:
```suggestion
  return socialTargetsDiff
    .slice()
    .sort(({ value: leftValue }, { value: rightValue }) =>
      stringCompareFunction(leftValue, rightValue)
    );
```
Can be simplified to:
```ts
return socialTargetsDiff.slice().sort(stringCompareFunction);
```

File: packages/console/src/pages/SignInExperience/components/SignInMethodsChangePreview.module.scss [hunk 2]
```diff
@@ -5,24 +5,26 @@
...
   font: var(--font-body-medium);
+  justify-content: space-between;
+  align-items: stretch;
+  width: 552px;
 
   .section {
-    &:not(:first-child) {
-      margin-top: _.unit(3);
-    }
+    width: 270px;
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsChangePreview.module.scss]: updated.

File: packages/console/src/pages/SignInExperience/components/SignInMethodsChangePreview.module.scss [hunk 3]
```diff
@@ -5,24 +5,26 @@
...
+  display: flex;
+  flex-direction: row;
   margin-top: _.unit(6);
   border-radius: 8px;
-  padding: _.unit(5);
-  background: var(--color-layer-2);
   font: var(--font-body-medium);
+  justify-content: space-between;
+  align-items: stretch;
+  width: 552px;
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsChangePreview.module.scss]: updated.

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/utilities.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/utilities.ts]: No, the params of the sort function are objects, not just simple strings.

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/DiffSegment.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+type Props = {
+  children: ReactNode;
+  hasChanged: boolean;
+  isAfter?: boolean;
+};
+
+const DiffSegment = ({ children, hasChanged, isAfter = false }: Props) => {
+  if (!hasChanged) {
+    // eslint-disable-next-line react/jsx-no-useless-fragment
+    return <>{children}</>;
```
Comment (simeng-li) [inline packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/DiffSegment.tsx:14]: will return children do? not sure

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/SignInDiffSection.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+const SignInDiffSection = ({ before, after, isAfter = false }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const beforeSignInMethodsObject = convertToSignInMethodsObject(
+    before
+      .slice()
+      .sort(({ identifier: leftIdentifier }, { identifier: rightIdentifier }) =>
+        stringCompareFunction(leftIdentifier, rightIdentifier)
+      )
+  );
```
Comment (simeng-li) [inline packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/SignInDiffSection.tsx:29]: no order in object

**Review by simeng-li — APPROVED** [review]

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/DiffSegment.tsx (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/DiffSegment.tsx:14]: If we return children, we'll get an error:
image

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/SignInDiffSection.tsx (hunk 5 above)
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/SignInDiffSection.tsx]: Updated. Sort when rendering.

**Pull request merged**